### DISCUSSION TITLE
the buttons are not responding

### DISCUSSION THREAD

**Discussion by jendass** [kachurovskiy/nanoels#17] [Q&A]
Hi, everybody. 
Thanks  for the excellent project.
My control panel does not respond to buttons if I have it powered from a 5v transformer. But if I connect the panel to a mini usb power supply the buttons respond.
Do you know what could be causing this? Thanks

**Comment by kachurovskiy** [comment]
jendass, please check the continuity between Arduino ground and the ground you use when "powered from a 5v transformer". E.g. Arduino USB port metal housing should be connected to all other GND points on the board.

Apart from that, checking/re-melting all solder joints on the board might be worth a try.

Could you post a photo of your PCB assembly (top and bottom)? Many thanks.

**Comment by jendass** [comment]
![2021-12-02 10 39 45](https://user-images.githubusercontent.com/26498160/144397006-86b1ec30-64e0-4fe3-8ac6-bb80f301ed3d.jpg)
![2021-12-02 10 40 14](https://user-images.githubusercontent.com/26498160/144397017-9c819c40-cd37-4e19-ae15-aa08637a208a.jpg)
thanks for the quick response.
Every GND is connected with Arduino  USB case

**Reply by R2-66** [reply]
Hi Jendass it looks like some pins on the board don't have enough solder, as Maxim suggested I would go through and resolder all connections on the pcb...good luck and keep us posted

**Reply by kachurovskiy** [reply]
As you can see from schematics, buttons work by shorting Arduino pins to ground.

<img width="1052" alt="nanoels-schematics" src="https://user-images.githubusercontent.com/517919/144491680-640ed245-de44-4ce1-87c7-597da0abed10.png">

If they work in one case but not in the other, the problem is with grounding of the circuit.

Going through some/all of the spots where you soldered, touching them with the soldering iron to make the solder liquid again, might help - in case there's a bad contact somewhere. Pay special attention to all the GND connections.

Alternatively this could be caused by different grounding of your power sources, electrical engineering is not my strength :)

Another reason could be a partially faulty Arduino - sadly this happens more often than you'd expect with 3$ Arduinos - e.g. some traces inside the board are lost/messed up.

**Reply by Hedning003** [reply]
i had the same problem when i fiddled around on the bench - i check'd all the solderingpoints whit a magnifying glas,and had to resolder 3 pins on the nano-that fixed the problem for me

**Comment by jendass** [comment]
I resolder all contacts twice. tried other 5v power supplies. reinstalled Arduino. I'm soldering it under a magnifying glass, at 35 years old I can't see it anymore. I'll try to buy the original Arduino and new buttons and start over. Thanks, everyone.

**Reply by kachurovskiy** [reply]
Hey jendass, wait a bit, let's ask on https://electronics.stackexchange.com/questions/597663/to-ground-buttons-only-working-when-arduino-is-powered-via-usb and see if that helps first.

**Comment by jendass** [comment]
I find that once I connect the board to the power supply, I can press the button once and it responds(sometimes). Then it doesn't. Thanks

**Reply by kachurovskiy** [reply]
As you can see in https://github.com/kachurovskiy/nanoels/blob/main/h1/NanoEls.ino#L323 we're pulling up (to 5V) all button pins so the fact that it only works once means that the pin is not sufficiently pulled up - maybe there's a problem with the built-in resistors in that Arduino? Or some medium-resistance connection exists between pins and ground which pulls them under 3V which is the limit for `digitalRead` - e.g. through that flux residue layer that is visible on your photos?

**Comment by kachurovskiy** [comment]
jendass, please check replies on https://electronics.stackexchange.com/questions/597663/to-ground-buttons-only-working-when-arduino-is-powered-via-usb and let us know if it helps

**Comment by kachurovskiy** [comment]
Kind folks on Electrical Engineering have brought to my attention that many buttons are not directly connected to GND layer on this version of the PCB - instead, they are connected via other buttons. This have worked for most of us but maybe your buttons are a bit special or something else is at play. A possible fix would be to connect 1 top leg of non-working buttons to GND somewhere on the board with a small cable.

I have a much improved version of the PCB designed in EasyEDA but I didn't post it online because I didn't test it in the real world. Doing that could fix this for the future builders.

**Reply by jendass** [reply]
I don't have much experience with it. I'll try my best next week as I have more time. Anyway, thank you for your time and effort to find out where the problem is.

**Comment by ws786873** [comment]
I saw the answer of a kind person. Maybe you can disconnect all GND pins and connect all GND together by using the copper clad function. Then he also mentioned the use of external pull-up resistor. You can connect a 4.7K resistor in series to each switch. Of course, you can use 0805 package, which won't take up too much space. My knowledge is limited and I just put forward some suggestions that may be useful.Of course, all this work will be done after jendass solves the problem by using flying wires to connect GND or series resistors.

**Comment by mockendon** [comment]
Each button passes the ground to the next one via internal continuity (across the legs) one defective switch in the chain could cause all to stop working down stream.  Perhaps put a continuity tester across the switches to confirm that they are intact.

**Reply by kachurovskiy** [reply]
Yeah, that was the weak point in 2020-12-25 PCB version. This should no longer be a risk in latest PCB version from 2021-12-04.
